### PR TITLE
Feat/colors palette

### PR DESCRIPTION
**Pull request by fredericvilcot**
This PR implements the colors palette defined at OKP4 and resolves okp4/product#412.

There is a lot of naming conventions for colors, so I suggested to define names according to [Geoff Graham](https://geoffgraham.me/) guidelines, which seems to be smart for our design system. (👉 variables grouped by color)

👉 About `stylelint` rules : 
- [color-named](https://stylelint.io/user-guide/rules/list/color-named) has been kept enabled but soften only for `$color-pure-white` variable to keep exposing the hexadecimal value in associated story
- [color-no-invalid-hex](https://stylelint.io/user-guide/rules/list/color-no-invalid-hex) has been enabled to prevent misuse of hexadecimal values

OKP4 palette : 
Capture d’écran 2022-03-07 à 12 24 40

### CODE REVIEW

**Comment by ccamel** [comment]
It seems that the rule `color-named`, specified in [stylelintrc.json](https://github.com/okp4/ui/blob/main/.stylelintrc.json#L7) is not really suitable for our needs.

**Review by ccamel — CHANGES_REQUESTED** [review]
Nice work. :+1: 

I'd suggest to move all the things related to the stories into the folder `stories`. The folder `src` should only contain the source code of the library.

**Review by amimart — APPROVED** [review]

**Review by nperon — APPROVED** [review]

**Comment by fredericvilcot** [comment]
@ccamel I suggest not to implement more colors and gradients in the palette because Vincent continues to experiment with new colors to determine the visual identity of okp4...

**Comment by ccamel** [comment]
Ok. Fine for me.

**Review by ad2ien — DISMISSED** [review]

**Review by ccamel — APPROVED** [review]

File: stories/brand/colors/Color.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { ColorKey, colorName, colorVariable } from './utils'
+import colors from './_exports.module.scss'
+
+export type StorybookColorProps = {
+    /**
+     * References the key of expoted the variable in the color palette.
```
Comment (amimart) [inline stories/brand/colors/Color.stories.tsx:7]:
```suggestion
     * References the key of the exported variable in the color palette.
```

Little typo ? 👨‍🏫

**Review by amimart — CHANGES_REQUESTED** [review]

File: stories/brand/colors/Color.stories.tsx (hunk 1 above)
Comment (fredericvilcot) [reply stories/brand/colors/Color.stories.tsx:7]: Ahahah yep, I'll make the change 😜

**Comment by fredericvilcot** [comment]
Everything has been cleaned, if it's ok for you, let's merge it 🚀

**Review by amimart — APPROVED** [review]

**Review by ccamel — APPROVED** [review]

**Pull request merged**